src/environment.js

```diff
@@ -208,11 +214,18 @@ var Environment = Obj.extend({
                     }
                 }
                 else {
-                    var tmpl = new Template(info.src, _this,
+                    var tmpl;
+                    if(ignoreMissing && fileMissing) {
```

Comment:
there's no need to check `ignoreMissing` again here -- if the template is missing and `ignoreMissing` is not set, there'd have already been an error.